### PR TITLE
Example demonstrating the use of Texture subimages.

### PR DESCRIPTION
**Pull request by rogermc2**
This example is intended to demonstrate the use of Texture_2D.Storage (glTexStorage2D) and 
glTexSubImage2D.
As glTexSubImage2D does not seem to have been previously implemented, I have provided a new Texture_2D.Load_SubImage_From_Data procedure following the generate instructions provided in the Developer Documentation section of the OpenGLAda Readme.
Following review of my implementation, if accepted, I could implement corresponding code for glTexSubImage1D and glTexSubImage3D.
I have also provided an implementation for glIsTexture which I have found useful for debugging in a another project.

### CODE REVIEW

File: src/gl/implementation/gl-objects-textures-with_2d_loader.adb [hunk 1]
```diff
@@ -47,6 +47,20 @@ package body GL.Objects.Textures.With_2D_Loader is
       Raise_Exception_On_OpenGL_Error;
    end Load_From_Data;
 
+   procedure Load_SubImage_From_Data
```
Comment (flyx) [inline src/gl/implementation/gl-objects-textures-with_2d_loader.adb]: I'd like to have this changed to `Load_Sub_Image_From_Data` so it's consistent with all other names.

Comment (rogermc2) [reply src/gl/implementation/gl-objects-textures-with_2d_loader.adb]: Fine.
Should I implement corresponding Load_Sub_Image_From_Data for glTexSubImage1D and glTexSubImage3D?

Comment (flyx) [reply src/gl/implementation/gl-objects-textures-with_2d_loader.adb]: Would certainly be good to have them, please do.

Comment (rogermc2) [reply src/gl/implementation/gl-objects-textures-with_2d_loader.adb]: OK, tomorrow's job.

**Comment by rogermc2** [comment]
Requested changes implemented.

**Pull request merged**